sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/ListingFileCatalog.scala

```diff
@@ -30,6 +29,13 @@ import org.apache.spark.sql.types.StructType
 
 
 /**
+ * A filter class to list up qualified files in parallel.
+ */
+private[spark] abstract class SerializablePathFilter extends Serializable {
```

Comment:
Also, it probably makes more sense to move this class into `fileSourceInterfaces.scala` since it's part of the public interface.